app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java

```diff
@@ -0,0 +1,109 @@
+package cs5150athletetracking.com.athletetracking;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import java.io.*;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.net.URLEncoder;
+import java.util.Iterator;
+
+import org.json.JSONObject;
+
+
+/**
+ * Created by mag94 on 10/19/15.
+ */
+
+public class AsyncUploader extends AsyncTask<LocationJSON, String, Integer> {
+
+    private static final String TAG = "Http Connection";
+    private static final String URL_STRING = "http://ec2-54-165-208-160.compute-1.amazonaws.com/php/test_server.php";
+    private static final int END_OF_STREAM = -1;
+    private static final int HTTP_OK = 200;
+    private static final int SUCCESS = 1;
+    private static final int FAILURE = -1;
+
+    private ResultCallable callBack;
+
+    public void setCallBack(ResultCallable callBack) {
+        this.callBack = callBack;
+    }
+
+    public ResultCallable getCallBack() {
+        return callBack;
+    }
+
+    @Override
+    protected Integer doInBackground(LocationJSON... params) {
+        Integer result = FAILURE;
+
+        try {
+            int count = params.length;
+            for (int i = 0; i < count; i++) {
+                LocationJSON param = params[i];
+                StringBuilder jsonPost = new StringBuilder();
+                Iterator<String> keys = param.keys();
+
+                // Format LocationJSON argument for the POST method
+                // The format is: "key1=value1&key2=value2"
+                while (keys.hasNext()) {
```

Comment:
Rather than do all of this, which only looks like it will open the first level of the JSON, why not just get the toString() representation of the entire JSON and upload that (by writing to the output stream). Any half-good JSON library on the backend will be able to parse a JSON with arbitrary structure from a string representation.